software/android_app/app/src/main/java/com/example/signbuddy/LessonActivity.java

```diff
@@ -1,47 +1,235 @@
+    private int msg_length;
+    private int skip_crc;
+    private boolean next_byte_is_length;
+    private String gestureLetter = "";
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_lesson);
-
-        listView = findViewById(R.id.listView);
-
-        //array adapter
-        //populate data from array into listView
-        //takes context of R.layout, layout/arrangement of data, data to be displayed
-        ArrayAdapter adapter = new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, categoryTypes);
-
-        listView.setAdapter(adapter); //link adapter to listView
-
-        //event handler for clickEvents
-        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
-            @Override
-            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
-                if (position == 0) {
-                    //clicked alphabet
-                    //go from main activity to alphabet activity
-                    startActivity(new Intent(LessonActivity.this, AlphabetActivity.class));
-                } else {
+        setContentView(R.layout.activity_alesson);
+        ImageView image = (ImageView) findViewById(R.id.signImage);
+
+        //setting gesture image
+        Intent i = getIntent();
+        String lessonNumber = i.getStringExtra("LessonNumber");
+        lessonnum = Integer.valueOf(lessonNumber);
+        alpha = i.getIntExtra("alpha", 0);
+
+        //set Progress Bar
+        pgsBar = (ProgressBar) findViewById(R.id.progressBar);
+        pgsBar.setProgress(alpha);
+
+        //setting current letter
+        if (alpha ==5){
+            finish();
+            Toast.makeText(LessonActivity.this, "You have completed Lesson " + lessonnum, Toast.LENGTH_SHORT).show();
+            Intent intent = new Intent(LessonActivity.this, LessonMenuActivity.class);
+            startActivity(intent);
+
+        }
+
+        //setting current image
+        alphanum  = (lessonnum-1)*5+alpha;
+        String alphaNumber = String.valueOf(alphanum);
+        String fnm = "alpha_"+alphaNumber; //  image file name
+        String PACKAGE_NAME = getApplicationContext().getPackageName();
+        int imgId = getResources().getIdentifier(PACKAGE_NAME+":drawable/"+fnm , null, null);
+        image.setImageResource(imgId);
+
+        //start countdown
+        new CountDownTimer(7500, 1500) {
+            public void onTick(long millisUntilFinished) {
+                countText = findViewById(R.id.countText);
+                countText.setText("Perform Sign in: " + ((millisUntilFinished / 1500)+1));
+            }
+            public void onFinish() {
+                countText.setText("GO!");
+                BleManager.getInstance().notify(MainActivity.SignBuddy, UUID_SERVICE, UUID_CHARACTERISTIC_NOTIFY, new BleNotifyCallback() {
+                    @Override
```

Comment:
Notify should be called before this somewhere. This is where you actually need to send the command to the device to record the data, and so the app should already be waiting to be notified